### DISCUSSION TITLE
"ArgumentError: wrong number of arguments" after upgrade from Postal v2 to v3

### DISCUSSION THREAD

**Discussion by cdo-iban** [postalserver/postal#2883] [Help with using Postal] [answered]
Hi,
we've bee ntrying to upggrade Postal to v3 from v2 but mail delivery fails with error: 
![image](https://github.com/postalserver/postal/assets/119871428/75dbb943-2d67-4c43-b811-80d758fb3b62)

We've upgraded from v2.1.4 to v2.3.2 without issues, mails are sent as expected.
We then upgraded from v2.3.2 to v3.0.2 and got delivery failure for all the e-mails.
Same with v3.2.2.
We've rolled back to v2.3.2 and mail could be delivered again.

There is not much in the (docker) logs, and we've no experience whatsoever with Ruby/Rails. Could one tell where to look?

Cheers,
Claude

**Comment by willpower232** [comment]
Which docker logs did you check? I presume you would have been able to see a stack trace in the logs for the worker perhaps?

**Comment by adamcooke** [comment]
Please check your worker log with `postal logs worker` to see the full error here.

**Comment by cronax1234** [comment]
Same problem here, when using smtp_relays "ArgumentError: wrong number of arguments"

**Comment by adamcooke** [comment, marked as answer]
This is fixed in `3.3.1`. This was the first time someone mentioned SMTP Relays and the OP couldn't provide any logs to help pinpoint the issue 😞
